google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudLibraryDependencyWriter.java

```diff
@@ -121,23 +130,81 @@ protected void run(@NotNull Result result) throws Throwable {
         // user when it's complete.
         List<MavenId> newMavenIds = mavenIdsMap.get(false);
         if (!newMavenIds.isEmpty()) {
-          newMavenIds.forEach(
-              mavenId -> {
-                MavenDomUtil.createDomDependency(model, /* editor= */ null, mavenId);
-                UsageTrackerProvider.getInstance()
-                    .trackEvent(GctTracking.CLIENT_LIBRARY_ADD_LIBRARY)
-                    .addMetadata(
-                        GctTracking.METADATA_BUILD_SYSTEM_KEY,
-                        GctTracking.METADATA_BUILD_SYSTEM_MAVEN)
-                    .addMetadata(GctTracking.METADATA_LABEL_KEY, mavenId.getDisplayString())
-                    .ping();
-              });
+          newMavenIds.forEach(mavenId -> writeNewMavenDependency(model, mavenId, bomVersion));
+
+          if (bomVersion != null) {
+            addBomToMavenModule(model, bomVersion);
+          }
+
           notifyAddedDependencies(newMavenIds, project);
         }
       }
     }.execute();
   }
 
+  /**
+   * Adds the google-cloud-java BOM dependency to the module's dependencyManagement section.
+   *
+   * <p>If the BOM already exists, then this will update the version of the BOM to match the
+   * supplied version.
+   */
+  private static void addBomToMavenModule(MavenDomProjectModel model, String bomVersion) {
+    MavenDomDependencies mavenDomDependencies = model.getDependencyManagement().getDependencies();
+    Optional<MavenDomDependency> bomDependencyOptional =
+        mavenDomDependencies
+            .getDependencies()
+            .stream()
+            .filter(
+                mdd ->
+                    CloudApiMavenService.GOOGLE_CLOUD_JAVA_BOM_ARTIFACT.equals(
+                        mdd.getArtifactId().getStringValue()))
+            .findFirst();
+
+    if (!bomDependencyOptional.isPresent()) {
```

Comment:
I think you mentioned this for the next PR, +1 to unit test this logic.

Reply:
I'm going to take a stab at doing the refactoring for testability here and see if it distracts too much from this PR. that way we can get the tests in now together with these updates

Reply:
done - in this PR